frontend/src/common/navbar/index.tsx

```diff
@@ -1,111 +1,82 @@
 
-interface NavbarProps {
-  navItems: string[];
-}
+const NavBar = () => {
+  const isLoggedIn = useIsLoggedIn();
+  const [performLogOut, logOutResult] = useLogOut();
 
-interface DrawerProps {
-  navItems: string[];
-  handleDrawerToggle: () => void;
-}
+  const [authModalOpen, setAuthModalOpen] = useState(false); // added this state
 
-const drawerWidth = 240;
-
-const DrawerElement = ({ handleDrawerToggle, navItems }: DrawerProps) => (
-  <Box onClick={handleDrawerToggle} sx={{ textAlign: 'center' }}>
-    <Typography variant="h6" sx={{ my: 2 }}>
-      MUI
-    </Typography>
-    <Divider />
-    <List>
-      {navItems.map((item: string) => (
-        <ListItem key={item} disablePadding>
-          <ListItemButton sx={{ textAlign: 'center' }}>
-            <ListItemText primary={item} />
-          </ListItemButton>
-        </ListItem>
-      ))}
-    </List>
-  </Box>
-);
-
-const NavBar = ({ navItems }: NavbarProps) => {
-  // const window = new Window();
-  const [mobileOpen, setMobileOpen] = React.useState(false);
-  const loggedIn = useIsLoggedIn();
-
-  console.log('logged in? ', loggedIn); //FOR DONGYI -> loggedIn == true -> user is logged in -> show profile, dashboard buttons etc.
-
-  const handleDrawerToggle = () => {
-    setMobileOpen((prevState) => !prevState);
+  const handleLogOut = async (event: React.MouseEvent) => {
+    event.preventDefault(); // prevent default anchor behavior
+    await performLogOut();
+    window.location.href = '/'; // redirect to home after logout
   };
 
-  const container =
-    window !== undefined ? () => window.document.body : undefined;
-
   return (
-    <div style={{ position: 'initial' }}>
-      <AppBar component="nav" position="sticky">
+    <div>
+      <AppBar position="sticky">
         <Toolbar>
-          <IconButton
-            color="inherit"
-            aria-label="open drawer"
-            edge="start"
-            onClick={handleDrawerToggle}
-            sx={{ mr: 2, display: { sm: 'none' } }}
-          >
-            <MenuIcon />
-          </IconButton>
-          <Typography
-            variant="h6"
-            component="div"
-            sx={{ flexGrow: 1, display: { xs: 'none', sm: 'block' } }}
-          >
-            MUI
+          <Typography variant="h6" style={{ flexGrow: 1 }}>
+            KanBan
           </Typography>
-          <Box sx={{ display: { xs: 'none', sm: 'block' } }}>
-            {navItems.map((item) => (
-              <Button key={item} sx={{ color: '#fff' }}>
-                {item}
+
+          {/* If the user is logged in, show dashboard, board, and profile links */}
+          {isLoggedIn && (
+            <>
+              <a href="/" style={{ textDecoration: 'none' }}>
+                <Button color="inherit">
+                  Home
+                </Button>
+              </a>
+              <a href="/dashboard" style={{ textDecoration: 'none' }}>
+                <Button color="inherit">
```

Comment:
Just another way to write the code could be:

import {useNavigate} from 'react-router-dom';
const navigateTo = useNavigate();

....
.....
....
<Button color="inherit" onClick={() => navigateTo(Routes.Profile)}>
    Profile
</Button>

Reply:
I used the Navigate Function but it didnt work, instead it will become a empty page, thats why I use the hardcode.
And route works, I can replace the href with the route part.

Reply:
> I used the Navigate Function but it didnt work, instead it will become a empty page, thats why I use the hardcode. And route works, I can replace the href with the route part.

can you show an example of what you were trying?

Reply:
a screenshot would be best

Reply:
import { useNavigate } from 'react-router-dom';

interface NavbarProps {
  navItems: string[];
}

const NavBar = ({ navItems }: NavbarProps) => {
  const navigate = useNavigate();
  
  ......
  
  
           <Button color="inherit" onClick={() => navigator("/dashboard")}>
            Dashboard
          </Button>
          <Button color="inherit" onClick={() => navigator("/board")}>
            Board
          </Button>
          <Button color="inherit" onClick={() => navigator("/profile")}>
            Profile
          </Button>